### ISSUE TITLE
Polyline/polygon bounding box cuts off part of miter joint

### ISSUE THREAD

**Issue by ANSH3LL** [yisibl/resvg-js#305]
The tips of miter joints in polygons, polylines and paths are cut-off when `Crop by BBox` is selected.
Rect shapes rotated 45 degrees are also affected when stroke is applied.
It seems that the bbox calculations do not include the full length of the miter.
This might be a bug in the underlying `resvg` library, but I'm not sure how `resvg-js` performs cropping.

<details>
    <summary>Polyline: Before cropping</summary>

![polyline-not-cropped](https://github.com/yisibl/resvg-js/assets/53097224/ff82cc61-76dd-49fa-86c8-be6a3e94ef22)
</details>

<details>
    <summary>Polyline: After cropping</summary>

![polyline-cropped](https://github.com/yisibl/resvg-js/assets/53097224/0cce3842-cb67-4804-ae3f-6924b014bd54)
</details>

<details>
    <summary>Path: Before cropping</summary>

![path-not-cropped](https://github.com/yisibl/resvg-js/assets/53097224/cd269be2-b916-4646-b424-286de024276d)
</details>

<details>
    <summary>Path: After cropping</summary>

![path-cropped](https://github.com/yisibl/resvg-js/assets/53097224/4e433535-00dd-4767-b536-c15fffd7fea3)
</details>

<details>
    <summary>Rotated rect: Before cropping</summary>

![rect-not-cropped](https://github.com/yisibl/resvg-js/assets/53097224/a6dbbe95-33fe-44ff-a60f-f630203889e8)
</details>

<details>
    <summary>Rotated rect: After cropping</summary>

![rect-cropped](https://github.com/yisibl/resvg-js/assets/53097224/9b8de059-5298-4c7a-923b-fe370e802aaa)
</details>

<hr>

<details>
    <summary>Polyline svg</summary>

```xml
<svg viewBox="18 8 43 24" version="1.1" xmlns="http://www.w3.org/2000/svg">
    <polyline points="60 20 20 20 30 30 30 10 40 20" fill="none" stroke="black" stroke-linejoin="miter" />
</svg>
```
</details>
<details>
    <summary>Path svg</summary>

```xml
<svg viewBox="0 0 38 30" xmlns="http://www.w3.org/2000/svg">
  <path
    stroke="black"
    fill="none"
    stroke-linejoin="miter"
    stroke-miterlimit="8"
    d="M1,29 l7   ,-3 l7   ,3
       m2, 0 l3.5 ,-3 l3.5 ,3
       m2, 0 l2   ,-3 l2   ,3
       m2, 0 l0.75,-3 l0.75,3
       m2, 0 l0.5 ,-3 l0.5 ,3" />
</svg>
```
</details>
<details>
    <summary>Rect svg</summary>

```xml
<svg width="31" height="31" xmlns="http://www.w3.org/2000/svg">
  <rect x="12" y="-10" width="20" height="20" transform="rotate(45)" stroke="pink" stroke-width="2" />
</svg>
```
</details>

**Comment by yisibl** [comment]
Yes, it could be an upstream problem. You need to clone the latest resvg and run:

```
cargo run --example draw_bboxes -- bbox.svg bbox.png -z 10
```
You'll see that the bbox has a red border.

By the way, in what scenarios do you use Crop by BBox?

**Comment by ANSH3LL** [comment]
I use it when generating shapes programmatically so that I don't have to calculate the svg viewbox.

**Comment by ANSH3LL** [comment]
It appears that the problem is in resvg. The polyline svg's bbox is especially broken. Maybe @RazrFalcon can look into this?

![s](https://github.com/yisibl/resvg-js/assets/53097224/28a3bab0-0de0-4646-be36-d6f08b85240d)

![s2](https://github.com/yisibl/resvg-js/assets/53097224/3c18fdad-badf-4d06-8d55-fb4a383f49bb)

![s3](https://github.com/yisibl/resvg-js/assets/53097224/610e52f4-ea23-4c01-a7d0-22d91b2505fd)

**Comment by RazrFalcon** [comment]
Path bbox calculation is correct. The `draw_bbox` example glitches out on `Polyline svg` because it ignores the viewbox transform. I've updated the example code to fix it just now.

Red rectangle - path bbox.
Green rectangle - path bbox including stroke.

Meaning the issue is with the way resvg-js reinterprets resvg/usvg output.

**Comment by yisibl** [comment]
> Meaning the issue is with the way resvg-js reinterprets resvg/usvg output.

Yes, This requires resvg-js to update resvg to the latest version.
